### PR TITLE
Add members_count in user_groups/member_of api

### PR DESCRIPTION
**Pull request by k9845**
Addresses 
- New Project Flow: User Group issues(https://zube.io/deep/deep/c/3724)
## Changes
 - Added members_count in `api/v1/user-groups/member-of/?user={}` api

Mention related users here if any.

## This PR doesn't introduce any:

- [x] temporary files, auto-generated files or secret keys
- [x] n+1 queries
- [x] flake8 issues
- [x] `print`
- [x] typos
- [x] unwanted comments

## This PR contains valid:

- [x] tests
- [ ] permission checks (tests here too)
- [ ] translations

### CODE REVIEW

File: apps/user_group/models.py [hunk 1]
```diff
@@ -33,9 +33,9 @@ def get_for(user):
 
     @staticmethod
     def get_for_member(user):
-        return UserGroup.objects.filter(
-            members=user
-        ).distinct()
+        return UserGroup.objects.annotate(
+            members_count=models.Count('members', distinct=True)
+        ).filter(members=user).distinct()
```
Comment (thenav56) [inline apps/user_group/models.py]: Let's annote in the views.

File: apps/user_group/models.py [hunk 2]
```diff
@@ -33,9 +33,9 @@ def get_for(user):
 
     @staticmethod
     def get_for_member(user):
-        return UserGroup.objects.filter(
-            members=user
-        ).distinct()
+        return UserGroup.objects.annotate(
+            members_count=models.Count('members', distinct=True)
```
Comment (AdityaKhatri) [inline apps/user_group/models.py]: Was members not used before?

**Review by AdityaKhatri — CHANGES_REQUESTED** [review]

File: apps/user_group/models.py (hunk 2 above)
Comment (thenav56) [reply apps/user_group/models.py]: Not sure what you meant. But the filter is done by `.filter(members=user)`. @k9845 is only adding annotate here (members_count).

Comment (AdityaKhatri) [reply apps/user_group/models.py]: I too am not sure what I mean, but just making sure this doesn't break the previous API usage.

Comment (thenav56) [reply apps/user_group/models.py]: @k9845 Could you add a test if not present for the filter behavior?

**Review by thenav56 — APPROVED** [review]

File: apps/user_group/models.py (hunk 2 above)
Comment (k9845) [reply apps/user_group/models.py]: This is done @AdityaKhatri

**Review by AdityaKhatri — APPROVED** [review]

**Pull request merged**